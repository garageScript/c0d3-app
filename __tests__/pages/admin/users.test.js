import React from 'react'
import {
  render,
  waitForElementToBeRemoved,
  fireEvent
} from '@testing-library/react'
import { MockedProvider } from '@apollo/client/testing'
import '@testing-library/jest-dom'
import GET_APP from '../../../graphql/queries/getApp'
import dummyLessonData from '../../../__dummy__/lessonData'
import dummySessionData from '../../../__dummy__/sessionData'
import dummyAlertData from '../../../__dummy__/alertData'
import Users from '../../../pages/admin/users'
import ALL_USERS from '../../../graphql/queries/allUsers'

const dummyUsersData = [
  {
    cliToken: 'WRyzWxDc_DqR3Avi7xcGD',
    email: 'some@mail.com',
    id: '5',
    isAdmin: true,
    name: 'admin admin',
    username: 'admin'
  },
  {
    cliToken: 'acRy_p-HK9sTU8H1GfrJV',
    email: 'some@mail.com',
    id: '6',
    isAdmin: false,
    name: 'newbie newbie',
    username: 'newbie',
    id: '2'
  }
]
const mocks = [
  {
    request: { query: ALL_USERS },
    result: {
      data: {
        allUsers: dummyUsersData
      }
    }
  },
  {
    request: { query: GET_APP },
    result: {
      data: {
        session: dummySessionData,
        lessons: dummyLessonData,
        alerts: dummyAlertData
      }
    }
  }
]

describe('Users test', () => {
  test('Should be able to switch search options', async () => {
    const { container, getByRole, queryByText, getByLabelText } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Users />
      </MockedProvider>
    )
    await waitForElementToBeRemoved(() => queryByText('Loading...'))
    expect(container).toMatchSnapshot()
    const input = getByLabelText('Search By:')
    expect(input).toBeTruthy()
    fireEvent.change(input, {
      target: { value: 'admin' }
    })
    expect(container).toMatchSnapshot()
    const searchByName = getByRole('button', { name: 'Name' })
    fireEvent.click(searchByName)
    expect(container).toMatchSnapshot()
    const filterByAdmin = getByRole('button', { name: 'Admins' })
    fireEvent.click(filterByAdmin)
    expect(container).toMatchSnapshot()
  })
})
