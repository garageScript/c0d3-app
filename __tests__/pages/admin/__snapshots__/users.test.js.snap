// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users test Should be able to switch search options 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand text-primary font-weight-bold"
        href="/"
      >
        C0D3
      </a>
      <div
        id="navbarNav"
      >
        <div
          class="navbar-nav collapse navbar-collapse"
        >
          <div
            class="navbar-nav collapse navbar-collapse"
          >
            <a
              class="nav-item nav-link"
              href="/curriculum"
            >
              Curriculum
            </a>
            <a
              class="nav-item nav-link"
              href="https://github.com/garageScript/c0d3-app"
              rel="noopener noreferrer"
              target="_blank"
            >
              Repo
            </a>
            <a
              class="nav-item nav-link"
              href="/docs/learn"
            >
              Learn
            </a>
            <a
              class="nav-item nav-link"
              href="https://discord.gg/c0d3"
              rel="noopener noreferrer"
              target="_blank"
            >
              Community
            </a>
            <div
              class="dropdown"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle btn btn-none"
                type="button"
              >
                Admin
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="d-flex"
      >
        <a
          class="btn btn-secondary border overflow-hidden p-2 text-truncate"
          href="/profile/fakeusername"
        >
           fakeusername
        </a>
        <button
          class="btn border ml-2"
          style="color: rgb(41, 41, 41);"
        >
          Logout
        </button>
      </div>
    </div>
  </nav>
  <div
    class="container-md"
  >
    <div
      class="d-flex flex-column col-12"
    >
      <h1
        class="text-primary text-center font-weight-bold display-1"
      >
        Users
      </h1>
      <div
        class="mb-2"
      >
        <div
          class="d-flex flex-row"
        >
          <h4
            class="text-primary font-weight-bold mr-1"
          >
            <label
              for="search"
            >
              Search By:
            </label>
          </h4>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              ID
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn bg-primary"
              style="color: rgb(255, 255, 255);"
            >
              Username
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Name
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Email
            </button>
          </div>
        </div>
      </div>
      <input
        class="form-control"
        id="search"
        type="text"
      />
      <div
        class="mt-2 mb-2"
      >
        <div
          class="d-flex flex-row"
        >
          <h4
            class="text-primary font-weight-bold mr-1"
          >
            Filter By:
          </h4>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Admins
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Non-Admins
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn bg-primary"
              style="color: rgb(255, 255, 255);"
            >
              None
            </button>
          </div>
        </div>
      </div>
      <table
        class="table table-striped"
      >
        <thead>
          <tr
            class="text-center"
          >
            <th>
              ID
            </th>
            <th>
              Username
            </th>
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Admin
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="text-center"
          >
            <td
              class="align-middle"
            >
              5
            </td>
            <td
              class="align-middle"
            >
              admin
            </td>
            <td
              class="align-middle"
            >
              admin admin
            </td>
            <td
              class="align-middle"
            >
              some@mail.com
            </td>
            <td
              class="align-middle"
            >
              <button
                class="btn bg-danger"
                style="color: rgb(41, 41, 41);"
              >
                Remove
              </button>
            </td>
          </tr>
          <tr
            class="text-center"
          >
            <td
              class="align-middle"
            >
              2
            </td>
            <td
              class="align-middle"
            >
              newbie
            </td>
            <td
              class="align-middle"
            >
              newbie newbie
            </td>
            <td
              class="align-middle"
            >
              some@mail.com
            </td>
            <td
              class="align-middle"
            >
              <button
                class="btn bg-success"
                style="color: rgb(41, 41, 41);"
              >
                Add
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <footer
    class="mt-5 font-weight-light text-center pb-5 text-muted"
  >
    Â© Copyright 2021 GarageScript
  </footer>
</div>
`;

exports[`Users test Should be able to switch search options 2`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand text-primary font-weight-bold"
        href="/"
      >
        C0D3
      </a>
      <div
        id="navbarNav"
      >
        <div
          class="navbar-nav collapse navbar-collapse"
        >
          <div
            class="navbar-nav collapse navbar-collapse"
          >
            <a
              class="nav-item nav-link"
              href="/curriculum"
            >
              Curriculum
            </a>
            <a
              class="nav-item nav-link"
              href="https://github.com/garageScript/c0d3-app"
              rel="noopener noreferrer"
              target="_blank"
            >
              Repo
            </a>
            <a
              class="nav-item nav-link"
              href="/docs/learn"
            >
              Learn
            </a>
            <a
              class="nav-item nav-link"
              href="https://discord.gg/c0d3"
              rel="noopener noreferrer"
              target="_blank"
            >
              Community
            </a>
            <div
              class="dropdown"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle btn btn-none"
                type="button"
              >
                Admin
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="d-flex"
      >
        <a
          class="btn btn-secondary border overflow-hidden p-2 text-truncate"
          href="/profile/fakeusername"
        >
           fakeusername
        </a>
        <button
          class="btn border ml-2"
          style="color: rgb(41, 41, 41);"
        >
          Logout
        </button>
      </div>
    </div>
  </nav>
  <div
    class="container-md"
  >
    <div
      class="d-flex flex-column col-12"
    >
      <h1
        class="text-primary text-center font-weight-bold display-1"
      >
        Users
      </h1>
      <div
        class="mb-2"
      >
        <div
          class="d-flex flex-row"
        >
          <h4
            class="text-primary font-weight-bold mr-1"
          >
            <label
              for="search"
            >
              Search By:
            </label>
          </h4>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              ID
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn bg-primary"
              style="color: rgb(255, 255, 255);"
            >
              Username
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Name
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Email
            </button>
          </div>
        </div>
      </div>
      <input
        class="form-control"
        id="search"
        type="text"
      />
      <div
        class="mt-2 mb-2"
      >
        <div
          class="d-flex flex-row"
        >
          <h4
            class="text-primary font-weight-bold mr-1"
          >
            Filter By:
          </h4>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Admins
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Non-Admins
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn bg-primary"
              style="color: rgb(255, 255, 255);"
            >
              None
            </button>
          </div>
        </div>
      </div>
      <table
        class="table table-striped"
      >
        <thead>
          <tr
            class="text-center"
          >
            <th>
              ID
            </th>
            <th>
              Username
            </th>
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Admin
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="text-center"
          >
            <td
              class="align-middle"
            >
              5
            </td>
            <td
              class="align-middle"
            >
              admin
            </td>
            <td
              class="align-middle"
            >
              admin admin
            </td>
            <td
              class="align-middle"
            >
              some@mail.com
            </td>
            <td
              class="align-middle"
            >
              <button
                class="btn bg-danger"
                style="color: rgb(41, 41, 41);"
              >
                Remove
              </button>
            </td>
          </tr>
          <tr
            class="text-center"
          >
            <td
              class="align-middle"
            >
              2
            </td>
            <td
              class="align-middle"
            >
              newbie
            </td>
            <td
              class="align-middle"
            >
              newbie newbie
            </td>
            <td
              class="align-middle"
            >
              some@mail.com
            </td>
            <td
              class="align-middle"
            >
              <button
                class="btn bg-success"
                style="color: rgb(41, 41, 41);"
              >
                Add
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <footer
    class="mt-5 font-weight-light text-center pb-5 text-muted"
  >
    Â© Copyright 2021 GarageScript
  </footer>
</div>
`;

exports[`Users test Should be able to switch search options 3`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand text-primary font-weight-bold"
        href="/"
      >
        C0D3
      </a>
      <div
        id="navbarNav"
      >
        <div
          class="navbar-nav collapse navbar-collapse"
        >
          <div
            class="navbar-nav collapse navbar-collapse"
          >
            <a
              class="nav-item nav-link"
              href="/curriculum"
            >
              Curriculum
            </a>
            <a
              class="nav-item nav-link"
              href="https://github.com/garageScript/c0d3-app"
              rel="noopener noreferrer"
              target="_blank"
            >
              Repo
            </a>
            <a
              class="nav-item nav-link"
              href="/docs/learn"
            >
              Learn
            </a>
            <a
              class="nav-item nav-link"
              href="https://discord.gg/c0d3"
              rel="noopener noreferrer"
              target="_blank"
            >
              Community
            </a>
            <div
              class="dropdown"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle btn btn-none"
                type="button"
              >
                Admin
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="d-flex"
      >
        <a
          class="btn btn-secondary border overflow-hidden p-2 text-truncate"
          href="/profile/fakeusername"
        >
           fakeusername
        </a>
        <button
          class="btn border ml-2"
          style="color: rgb(41, 41, 41);"
        >
          Logout
        </button>
      </div>
    </div>
  </nav>
  <div
    class="container-md"
  >
    <div
      class="d-flex flex-column col-12"
    >
      <h1
        class="text-primary text-center font-weight-bold display-1"
      >
        Users
      </h1>
      <div
        class="mb-2"
      >
        <div
          class="d-flex flex-row"
        >
          <h4
            class="text-primary font-weight-bold mr-1"
          >
            <label
              for="search"
            >
              Search By:
            </label>
          </h4>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              ID
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Username
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn bg-primary"
              style="color: rgb(255, 255, 255);"
            >
              Name
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Email
            </button>
          </div>
        </div>
      </div>
      <input
        class="form-control"
        id="search"
        type="text"
      />
      <div
        class="mt-2 mb-2"
      >
        <div
          class="d-flex flex-row"
        >
          <h4
            class="text-primary font-weight-bold mr-1"
          >
            Filter By:
          </h4>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Admins
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Non-Admins
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn bg-primary"
              style="color: rgb(255, 255, 255);"
            >
              None
            </button>
          </div>
        </div>
      </div>
      <table
        class="table table-striped"
      >
        <thead>
          <tr
            class="text-center"
          >
            <th>
              ID
            </th>
            <th>
              Username
            </th>
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Admin
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="text-center"
          >
            <td
              class="align-middle"
            >
              5
            </td>
            <td
              class="align-middle"
            >
              admin
            </td>
            <td
              class="align-middle"
            >
              admin admin
            </td>
            <td
              class="align-middle"
            >
              some@mail.com
            </td>
            <td
              class="align-middle"
            >
              <button
                class="btn bg-danger"
                style="color: rgb(41, 41, 41);"
              >
                Remove
              </button>
            </td>
          </tr>
          <tr
            class="text-center"
          >
            <td
              class="align-middle"
            >
              2
            </td>
            <td
              class="align-middle"
            >
              newbie
            </td>
            <td
              class="align-middle"
            >
              newbie newbie
            </td>
            <td
              class="align-middle"
            >
              some@mail.com
            </td>
            <td
              class="align-middle"
            >
              <button
                class="btn bg-success"
                style="color: rgb(41, 41, 41);"
              >
                Add
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <footer
    class="mt-5 font-weight-light text-center pb-5 text-muted"
  >
    Â© Copyright 2021 GarageScript
  </footer>
</div>
`;

exports[`Users test Should be able to switch search options 4`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand text-primary font-weight-bold"
        href="/"
      >
        C0D3
      </a>
      <div
        id="navbarNav"
      >
        <div
          class="navbar-nav collapse navbar-collapse"
        >
          <div
            class="navbar-nav collapse navbar-collapse"
          >
            <a
              class="nav-item nav-link"
              href="/curriculum"
            >
              Curriculum
            </a>
            <a
              class="nav-item nav-link"
              href="https://github.com/garageScript/c0d3-app"
              rel="noopener noreferrer"
              target="_blank"
            >
              Repo
            </a>
            <a
              class="nav-item nav-link"
              href="/docs/learn"
            >
              Learn
            </a>
            <a
              class="nav-item nav-link"
              href="https://discord.gg/c0d3"
              rel="noopener noreferrer"
              target="_blank"
            >
              Community
            </a>
            <div
              class="dropdown"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle btn btn-none"
                type="button"
              >
                Admin
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="d-flex"
      >
        <a
          class="btn btn-secondary border overflow-hidden p-2 text-truncate"
          href="/profile/fakeusername"
        >
           fakeusername
        </a>
        <button
          class="btn border ml-2"
          style="color: rgb(41, 41, 41);"
        >
          Logout
        </button>
      </div>
    </div>
  </nav>
  <div
    class="container-md"
  >
    <div
      class="d-flex flex-column col-12"
    >
      <h1
        class="text-primary text-center font-weight-bold display-1"
      >
        Users
      </h1>
      <div
        class="mb-2"
      >
        <div
          class="d-flex flex-row"
        >
          <h4
            class="text-primary font-weight-bold mr-1"
          >
            <label
              for="search"
            >
              Search By:
            </label>
          </h4>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              ID
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Username
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn bg-primary"
              style="color: rgb(255, 255, 255);"
            >
              Name
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Email
            </button>
          </div>
        </div>
      </div>
      <input
        class="form-control"
        id="search"
        type="text"
      />
      <div
        class="mt-2 mb-2"
      >
        <div
          class="d-flex flex-row"
        >
          <h4
            class="text-primary font-weight-bold mr-1"
          >
            Filter By:
          </h4>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn bg-primary"
              style="color: rgb(255, 255, 255);"
            >
              Admins
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              Non-Admins
            </button>
          </div>
          <div
            style="margin-top: -1.57px;"
          >
            <button
              class="btn"
              style="color: rgb(142, 142, 142);"
            >
              None
            </button>
          </div>
        </div>
      </div>
      <table
        class="table table-striped"
      >
        <thead>
          <tr
            class="text-center"
          >
            <th>
              ID
            </th>
            <th>
              Username
            </th>
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Admin
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="text-center"
          >
            <td
              class="align-middle"
            >
              5
            </td>
            <td
              class="align-middle"
            >
              admin
            </td>
            <td
              class="align-middle"
            >
              admin admin
            </td>
            <td
              class="align-middle"
            >
              some@mail.com
            </td>
            <td
              class="align-middle"
            >
              <button
                class="btn bg-danger"
                style="color: rgb(41, 41, 41);"
              >
                Remove
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <footer
    class="mt-5 font-weight-light text-center pb-5 text-muted"
  >
    Â© Copyright 2021 GarageScript
  </footer>
</div>
`;
